### PR TITLE
add Gradle compile + checkstyle + PMD workflow

### PR DESCRIPTION
**Pull request by JohnRillos**
- [ ] I have verified that this is not a new project. If it is, I have contacted the InfoSec team and completed their processes to gain proper approval from the following InfoSec team members:

- [ ] I have verified that the backout plan for this change conforms to our standard engineering backout plan located [here](https://payitdev.atlassian.net/wiki/spaces/SEC/pages/2833416205/Standard+Change+Control+Back+Out+Plan). If it does not, I have documented an alternative backout plan below:

- [ ] I have verified that this change is backwards compatible. If it is not, I have specified the breaking changes and how they will be handled below:

- [ ] I have verified that this change will not impact the security controls built into the application or introduce any new security vulnerabilities. If it will, I have defined the security impact below:

- [ ] I have verified that this change will not result in downtime. If it will, I have noted the impact below:

- [ ] I have verified that no new dependencies were introduced. If they were, I have vetted them below:

- [ ] I have verified that all applicable tests were updated to ensure complete test coverage of any new or modified code.

- [ ] I have verified that any relevant documentation such as the README is still up to date and not impacted by my changes. If documentation needs updating for accuracy, I have done so.

### CODE REVIEW

File: .github/workflows/gradle_build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          java-version: '13'
+          distribution: adopt
+      - name: Set up environment
+        run: | 
+          echo "ARTIFACTORY_URL=${{ secrets.ARTIFACTORY_URL }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_USERNAME=${{ secrets.ARTIFACTORY_READER_USERNAME }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_PASSWORD=${{ secrets.ARTIFACTORY_READER_PASSWORD }}" >> $GITHUB_ENV
+      - name: Compile
+        run: chmod +x gradlew && ./gradlew clean assemble --refresh-dependencies
```
Comment (w25r) [inline .github/workflows/gradle_build.yml:31]: Why clean and refresh-dependencies?

Also, should we compile tests too even though we're not running them?

File: .github/workflows/gradle_build.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        uses: actions/setup-java@v2
+        with:
+          java-version: '13'
+          distribution: adopt
+      - name: Set up environment
+        run: | 
+          echo "ARTIFACTORY_URL=${{ secrets.ARTIFACTORY_URL }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_USERNAME=${{ secrets.ARTIFACTORY_READER_USERNAME }}" >> $GITHUB_ENV
+          echo "ARTIFACTORY_READER_PASSWORD=${{ secrets.ARTIFACTORY_READER_PASSWORD }}" >> $GITHUB_ENV
+      - name: Compile
```
Comment (w25r) [inline .github/workflows/gradle_build.yml:30]: Did you look at the gradle action?  I think the only thing it probably buys us is some caching.  We can do that in v2 or v3 if we figure out how to do versions.

**Review by w25r — APPROVED** [review]

**Review by w25r — CHANGES_REQUESTED** [review]
I guess I can't approve this until the checklist is filled out

File: .github/workflows/gradle_build.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        run: chmod +x gradlew && ./gradlew clean assemble --refresh-dependencies
+      - name: Checkstyle
+        run: ./gradlew checkstyleMain checkstyleTest --stacktrace
+      - name: PMD
```
[the remark this answers is not in the document]
Comment (JohnRillos) [reply .github/workflows/gradle_build.yml:34]: We can change it if it's significantly faster.  That being said, I like having it granular like this

**Pull request closed without merging**